### ISSUE TITLE
fix: pdm.exceptions.InstallationError: Some package operations are not complete yet (pdm add -d seleninum)

### ISSUE THREAD

**Issue by jjangga0214** [pdm-project/pdm#728]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce

**pyproject.toml**:

```toml
[project]
name = ""
version = ""
description = ""
authors = [
    {name = "jjangga0214", email = "bnbcmindnpass@gmail.com"},
]
dependencies = []
requires-python = ">=3.9"
dynamic = ["classifiers"]
license = {text = "MIT"}

[project.urls]
homepage = ""

[project.optional-dependencies]
[build-system]
requires = ["pdm-pep517"]
build-backend = "pdm.pep517.api"

[tool]
[tool.pdm]
[tool.pdm.dev-dependencies]
dev = [
    "selenium~=4.0",
]
```

**command**:

```shell
$ pdm add -d selenium -v
Adding packages to dev dev-dependencies: selenium
======== Start resolving requirements ========
  selenium
  python>=3.9
  Adding requirement selenium
	Found matching candidates:
	  <Candidate selenium 4.0.0 from https://pypi.org/simple/selenium/>
	  <Candidate selenium 3.141.0 from https://pypi.org/simple/selenium/>
	  <Candidate selenium 3.14.1 from https://pypi.org/simple/selenium/>
	  <Candidate selenium 3.14.0 from https://pypi.org/simple/selenium/>
	  <Candidate selenium 3.13.0 from https://pypi.org/simple/selenium/>
	  <Candidate selenium 3.12.0 from https://pypi.org/simple/selenium/>
	  <Candidate selenium 3.11.0 from https://pypi.org/simple/selenium/>
	  <Candidate selenium 3.10.0 from https://pypi.org/simple/selenium/>
	  <Candidate selenium 3.9.0 from https://pypi.org/simple/selenium/>
	  <Candidate selenium 3.8.1 from https://pypi.org/simple/selenium/>
	  ... [144 more candidate(s)]
  Adding requirement python>=3.9
======== Starting round 0 ========
Resolving: new pin python >=3.9
======== Ending round 0 ========
======== Starting round 1 ========
  Adding requirement urllib3[secure]~=1.26(from selenium 4.0.0)
	Found matching candidates:
	  <Candidate urllib3 1.26.7 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.6 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.5 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.4 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.3 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.2 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.1 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.0 from https://pypi.org/simple/urllib3/>
  Adding requirement trio~=0.17(from selenium 4.0.0)
	Found matching candidates:
	  <Candidate trio 0.19.0 from https://pypi.org/simple/trio/>
	  <Candidate trio 0.18.0 from https://pypi.org/simple/trio/>
	  <Candidate trio 0.17.0 from https://pypi.org/simple/trio/>
  Adding requirement trio-websocket~=0.9(from selenium 4.0.0)
	Found matching candidates:
	  <Candidate trio-websocket 0.9.2 from https://pypi.org/simple/trio-websocket/>
	  <Candidate trio-websocket 0.9.1 from https://pypi.org/simple/trio-websocket/>
	  <Candidate trio-websocket 0.9.0 from https://pypi.org/simple/trio-websocket/>
  Adding requirement python>=3.7,<4.0(from selenium 4.0.0)
Resolving: new pin selenium 4.0.0
======== Ending round 1 ========
======== Starting round 2 ========
  Adding requirement attrs>=19.2.0(from trio 0.19.0)
	Found matching candidates:
	  <Candidate attrs 21.2.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 21.1.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 20.3.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 20.2.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 20.1.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 19.3.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 19.2.0 from https://pypi.org/simple/attrs/>
  Adding requirement sortedcontainers(from trio 0.19.0)
	Found matching candidates:
	  <Candidate sortedcontainers 2.4.0 from https://pypi.org/simple/sortedcontainers/>
	  <Candidate sortedcontainers 2.3.0 from https://pypi.org/simple/sortedcontainers/>
	  <Candidate sortedcontainers 2.2.2 from https://pypi.org/simple/sortedcontainers/>
	  <Candidate sortedcontainers 2.2.1 from https://pypi.org/simple/sortedcontainers/>
	  <Candidate sortedcontainers 2.2.0 from https://pypi.org/simple/sortedcontainers/>
	  <Candidate sortedcontainers 2.1.0 from https://pypi.org/simple/sortedcontainers/>
	  <Candidate sortedcontainers 2.0.5 from https://pypi.org/simple/sortedcontainers/>
	  <Candidate sortedcontainers 2.0.4 from https://pypi.org/simple/sortedcontainers/>
	  <Candidate sortedcontainers 2.0.3 from https://pypi.org/simple/sortedcontainers/>
	  <Candidate sortedcontainers 2.0.2 from https://pypi.org/simple/sortedcontainers/>
	  ... [51 more candidate(s)]
  Adding requirement async-generator>=1.9(from trio 0.19.0)
	Found matching candidates:
	  <Candidate async-generator 1.10 from https://pypi.org/simple/async-generator/>
	  <Candidate async-generator 1.9 from https://pypi.org/simple/async-generator/>
  Adding requirement idna(from trio 0.19.0)
	Found matching candidates:
	  <Candidate idna 3.3 from https://pypi.org/simple/idna/>
	  <Candidate idna 3.2 from https://pypi.org/simple/idna/>
	  <Candidate idna 3.1 from https://pypi.org/simple/idna/>
	  <Candidate idna 3.0 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.10 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.9 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.8 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.7 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.6 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.5 from https://pypi.org/simple/idna/>
	  ... [31 more candidate(s)]
  Adding requirement outcome(from trio 0.19.0)
	Found matching candidates:
	  <Candidate outcome 1.1.0 from https://pypi.org/simple/outcome/>
	  <Candidate outcome 1.0.1 from https://pypi.org/simple/outcome/>
	  <Candidate outcome 1.0.0 from https://pypi.org/simple/outcome/>
	  <Candidate outcome 0.1.0 from https://pypi.org/simple/outcome/>
  Adding requirement sniffio(from trio 0.19.0)
	Found matching candidates:
	  <Candidate sniffio 1.2.0 from https://pypi.org/simple/sniffio/>
	  <Candidate sniffio 1.1.0 from https://pypi.org/simple/sniffio/>
	  <Candidate sniffio 1.0.0 from https://pypi.org/simple/sniffio/>
	  <Candidate sniffio 0.0.0 from https://pypi.org/simple/sniffio/>
  Adding requirement cffi>=1.14; os_name == "nt" and implementation_name != "pypy"(from trio 0.19.0)
	Found matching candidates:
	  <Candidate cffi 1.15.0 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.6 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.5 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.4 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.3 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.2 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.1 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.0 from https://pypi.org/simple/cffi/>
  Adding requirement python>=3.6(from trio 0.19.0)
Resolving: new pin trio 0.19.0
======== Ending round 2 ========
======== Starting round 3 ========
  Adding requirement python>=3.5(from async-generator 1.10)
Resolving: new pin async-generator 1.10
======== Ending round 3 ========
======== Starting round 4 ========
  Adding requirement async-generator>=1.10(from trio-websocket 0.9.2)
	Found matching candidates:
	  <Candidate async-generator 1.10 from https://pypi.org/simple/async-generator/>
	  <Candidate async-generator 1.9 from https://pypi.org/simple/async-generator/>
  Adding requirement trio>=0.11(from trio-websocket 0.9.2)
	Found matching candidates:
	  <Candidate trio 0.19.0 from https://pypi.org/simple/trio/>
	  <Candidate trio 0.18.0 from https://pypi.org/simple/trio/>
	  <Candidate trio 0.17.0 from https://pypi.org/simple/trio/>
  Adding requirement wsproto>=0.14(from trio-websocket 0.9.2)
	Found matching candidates:
	  <Candidate wsproto 1.0.0 from https://pypi.org/simple/wsproto/>
	  <Candidate wsproto 0.15.0 from https://pypi.org/simple/wsproto/>
	  <Candidate wsproto 0.14.1 from https://pypi.org/simple/wsproto/>
	  <Candidate wsproto 0.14.0 from https://pypi.org/simple/wsproto/>
  Adding requirement python>=3.5(from trio-websocket 0.9.2)
Resolving: new pin trio-websocket 0.9.2
======== Ending round 4 ========
======== Starting round 5 ========
  Adding requirement python>=3.5(from sniffio 1.2.0)
Resolving: new pin sniffio 1.2.0
======== Ending round 5 ========
======== Starting round 6 ========
  Adding requirement attrs>=19.2.0(from outcome 1.1.0)
	Found matching candidates:
	  <Candidate attrs 21.2.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 21.1.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 20.3.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 20.2.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 20.1.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 19.3.0 from https://pypi.org/simple/attrs/>
	  <Candidate attrs 19.2.0 from https://pypi.org/simple/attrs/>
  Adding requirement python>=3.6(from outcome 1.1.0)
Resolving: new pin outcome 1.1.0
======== Ending round 6 ========
======== Starting round 7 ========
  Adding requirement h11<1,>=0.9.0(from wsproto 1.0.0)
	Found matching candidates:
	  <Candidate h11 0.12.0 from https://pypi.org/simple/h11/>
	  <Candidate h11 0.11.0 from https://pypi.org/simple/h11/>
	  <Candidate h11 0.10.0 from https://pypi.org/simple/h11/>
	  <Candidate h11 0.9.0 from https://pypi.org/simple/h11/>
  Adding requirement python>=3.6.1(from wsproto 1.0.0)
Resolving: new pin wsproto 1.0.0
======== Ending round 7 ========
======== Starting round 8 ========
  Adding requirement python>=3.6(from h11 0.12.0)
Resolving: new pin h11 0.12.0
======== Ending round 8 ========
======== Starting round 9 ========
  Adding requirement python>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*(from attrs 21.2.0)
Resolving: new pin attrs 21.2.0
======== Ending round 9 ========
======== Starting round 10 ========
  Adding requirement pyOpenSSL>=0.14(from urllib3 1.26.7)
	Found matching candidates:
	  <Candidate pyOpenSSL 21.0.0 from https://pypi.org/simple/pyopenssl/>
	  <Candidate pyOpenSSL 20.0.1 from https://pypi.org/simple/pyopenssl/>
	  <Candidate pyOpenSSL 20.0.0 from https://pypi.org/simple/pyopenssl/>
	  <Candidate pyOpenSSL 19.1.0 from https://pypi.org/simple/pyopenssl/>
	  <Candidate pyOpenSSL 19.0.0 from https://pypi.org/simple/pyopenssl/>
	  <Candidate pyOpenSSL 18.0.0 from https://pypi.org/simple/pyopenssl/>
	  <Candidate pyOpenSSL 17.5.0 from https://pypi.org/simple/pyopenssl/>
	  <Candidate pyOpenSSL 17.4.0 from https://pypi.org/simple/pyopenssl/>
	  <Candidate pyOpenSSL 17.3.0 from https://pypi.org/simple/pyopenssl/>
	  <Candidate pyOpenSSL 17.2.0 from https://pypi.org/simple/pyopenssl/>
	  ... [25 more candidate(s)]
  Adding requirement cryptography>=1.3.4(from urllib3 1.26.7)
	Found matching candidates:
	  <Candidate cryptography 35.0.0 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.8 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.7 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.6 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.5 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.4 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.3 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.2 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.1 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4 from https://pypi.org/simple/cryptography/>
	  ... [1102 more candidate(s)]
  Adding requirement idna>=2.0.0(from urllib3 1.26.7)
	Found matching candidates:
	  <Candidate idna 3.3 from https://pypi.org/simple/idna/>
	  <Candidate idna 3.2 from https://pypi.org/simple/idna/>
	  <Candidate idna 3.1 from https://pypi.org/simple/idna/>
	  <Candidate idna 3.0 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.10 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.9 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.8 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.7 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.6 from https://pypi.org/simple/idna/>
	  <Candidate idna 2.5 from https://pypi.org/simple/idna/>
	  ... [21 more candidate(s)]
  Adding requirement certifi(from urllib3 1.26.7)
	Found matching candidates:
	  <Candidate certifi 2021.10.8 from https://pypi.org/simple/certifi/>
	  <Candidate certifi 2021.5.30 from https://pypi.org/simple/certifi/>
	  <Candidate certifi 2020.12.5 from https://pypi.org/simple/certifi/>
	  <Candidate certifi 2020.11.8 from https://pypi.org/simple/certifi/>
	  <Candidate certifi 2020.6.20 from https://pypi.org/simple/certifi/>
	  <Candidate certifi 2020.4.5.2 from https://pypi.org/simple/certifi/>
	  <Candidate certifi 2020.4.5.1 from https://pypi.org/simple/certifi/>
	  <Candidate certifi 2020.4.5 from https://pypi.org/simple/certifi/>
	  <Candidate certifi 2019.11.28 from https://pypi.org/simple/certifi/>
	  <Candidate certifi 2019.9.11 from https://pypi.org/simple/certifi/>
	  ... [70 more candidate(s)]
  Adding requirement urllib3~=1.26(from urllib3 1.26.7)
	Found matching candidates:
	  <Candidate urllib3 1.26.7 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.6 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.5 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.4 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.3 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.2 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.1 from https://pypi.org/simple/urllib3/>
	  <Candidate urllib3 1.26.0 from https://pypi.org/simple/urllib3/>
  Adding requirement python>=2.7,<4.0,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*(from urllib3 1.26.7)
Resolving: new pin urllib3 1.26.7
======== Ending round 10 ========
======== Starting round 11 ========
  Adding requirement python>=2.7,<4.0,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*(from urllib3 1.26.7)
Resolving: new pin urllib3 1.26.7
======== Ending round 11 ========
======== Starting round 12 ========
  Adding requirement python>=3.5(from idna 3.3)
Resolving: new pin idna 3.3
======== Ending round 12 ========
======== Starting round 13 ========
  Adding requirement cryptography>=3.3(from pyOpenSSL 21.0.0)
	Found matching candidates:
	  <Candidate cryptography 35.0.0 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.8 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.7 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.6 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.5 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.4 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.3 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.2 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4.1 from https://pypi.org/simple/cryptography/>
	  <Candidate cryptography 3.4 from https://pypi.org/simple/cryptography/>
	  ... [1102 more candidate(s)]
  Adding requirement six>=1.5.2(from pyOpenSSL 21.0.0)
	Found matching candidates:
	  <Candidate six 1.16.0 from https://pypi.org/simple/six/>
	  <Candidate six 1.15.0 from https://pypi.org/simple/six/>
	  <Candidate six 1.14.0 from https://pypi.org/simple/six/>
	  <Candidate six 1.13.0 from https://pypi.org/simple/six/>
	  <Candidate six 1.12.0 from https://pypi.org/simple/six/>
	  <Candidate six 1.11.0 from https://pypi.org/simple/six/>
	  <Candidate six 1.10.0 from https://pypi.org/simple/six/>
	  <Candidate six 1.9.0 from https://pypi.org/simple/six/>
	  <Candidate six 1.8.0 from https://pypi.org/simple/six/>
	  <Candidate six 1.7.3 from https://pypi.org/simple/six/>
	  ... [22 more candidate(s)]
  Adding requirement python>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,!=3.5.*(from pyOpenSSL 21.0.0)
Resolving: new pin pyOpenSSL 21.0.0
======== Ending round 13 ========
======== Starting round 14 ========
  Adding requirement python>=2.7,!=3.0.*,!=3.1.*,!=3.2.*(from six 1.16.0)
Resolving: new pin six 1.16.0
======== Ending round 14 ========
======== Starting round 15 ========
Resolving: new pin sortedcontainers 2.4.0
======== Ending round 15 ========
======== Starting round 16 ========
Resolving: new pin certifi 2021.10.8
======== Ending round 16 ========
======== Starting round 17 ========
  Adding requirement cffi>=1.12(from cryptography 35.0.0)
	Found matching candidates:
	  <Candidate cffi 1.15.0 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.6 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.5 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.4 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.3 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.2 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.1 from https://pypi.org/simple/cffi/>
	  <Candidate cffi 1.14.0 from https://pypi.org/simple/cffi/>
  Adding requirement python>=3.6(from cryptography 35.0.0)
Resolving: new pin cryptography 35.0.0
======== Ending round 17 ========
======== Starting round 18 ========
  Adding requirement pycparser(from cffi 1.15.0)
	Found matching candidates:
	  <Candidate pycparser 2.21 from https://pypi.org/simple/pycparser/>
	  <Candidate pycparser 2.20 from https://pypi.org/simple/pycparser/>
	  <Candidate pycparser 2.19 from https://pypi.org/simple/pycparser/>
	  <Candidate pycparser 2.18 from https://pypi.org/simple/pycparser/>
	  <Candidate pycparser 2.17 from https://pypi.org/simple/pycparser/>
	  <Candidate pycparser 2.16 from https://pypi.org/simple/pycparser/>
	  <Candidate pycparser 2.15 from https://pypi.org/simple/pycparser/>
	  <Candidate pycparser 2.14 from https://pypi.org/simple/pycparser/>
	  <Candidate pycparser 2.13 from https://pypi.org/simple/pycparser/>
	  <Candidate pycparser 2.12 from https://pypi.org/simple/pycparser/>
	  ... [13 more candidate(s)]
Resolving: new pin cffi 1.15.0
======== Ending round 18 ========
======== Starting round 19 ========
Resolving: new pin pycparser 2.21
======== Ending round 19 ========
======== Starting round 20 ========
======== Resolution Result ========
Stable pins:
            python None
          selenium 4.0.0
              trio 0.19.0
   async-generator 1.10
    trio-websocket 0.9.2
           sniffio 1.2.0
           outcome 1.1.0
           wsproto 1.0.0
               h11 0.12.0
             attrs 21.2.0
   urllib3[secure] 1.26.7
           urllib3 1.26.7
              idna 3.3
         pyopenssl 21.0.0
               six 1.16.0
  sortedcontainers 2.4.0
           certifi 2021.10.8
      cryptography 35.0.0
              cffi 1.15.0
         pycparser 2.21
	Found matching candidates:
	  <Candidate selenium 4.0.0 from https://pypi.org/simple/selenium/>
	Found matching candidates:
	  <Candidate trio 0.19.0 from https://pypi.org/simple/trio/>
	Found matching candidates:
	  <Candidate async-generator 1.10 from https://pypi.org/simple/async-generator/>
	Found matching candidates:
	  <Candidate trio-websocket 0.9.2 from https://pypi.org/simple/trio-websocket/>
	Found matching candidates:
	  <Candidate sniffio 1.2.0 from https://pypi.org/simple/sniffio/>
	Found matching candidates:
	  <Candidate outcome 1.1.0 from https://pypi.org/simple/outcome/>
	Found matching candidates:
	  <Candidate wsproto 1.0.0 from https://pypi.org/simple/wsproto/>
	Found matching candidates:
	  <Candidate h11 0.12.0 from https://pypi.org/simple/h11/>
	Found matching candidates:
	  <Candidate attrs 21.2.0 from https://pypi.org/simple/attrs/>
	Found matching candidates:
	  <Candidate urllib3 1.26.7 from https://pypi.org/simple/urllib3/>
	Found matching candidates:
	  <Candidate urllib3 1.26.7 from https://pypi.org/simple/urllib3/>
	Found matching candidates:
	  <Candidate idna 3.3 from https://pypi.org/simple/idna/>
	Found matching candidates:
	  <Candidate pyOpenSSL 21.0.0 from https://pypi.org/simple/pyopenssl/>
	Found matching candidates:
	  <Candidate six 1.16.0 from https://pypi.org/simple/six/>
	Found matching candidates:
	  <Candidate sortedcontainers 2.4.0 from https://pypi.org/simple/sortedcontainers/>
	Found matching candidates:
	  <Candidate certifi 2021.10.8 from https://pypi.org/simple/certifi/>
	Found matching candidates:
	  <Candidate cryptography 35.0.0 from https://pypi.org/simple/cryptography/>
	Found matching candidates:
	  <Candidate cffi 1.15.0 from https://pypi.org/simple/cffi/>
	Found matching candidates:
	  <Candidate pycparser 2.21 from https://pypi.org/simple/pycparser/>
🔒 Lock successful
Changes are written to pdm.lock.
Changes are written to pyproject.toml.
Synchronizing working set with lock file: 18 to add, 0 to update, 0 to remove

Install attrs 21.2.0 successful
Install cffi 1.15.0 successful
Install certifi 2021.10.8 successful
Install h11 0.12.0 successful
Install async-generator 1.10 successful
Install idna 3.3 successful
Install outcome 1.1.0 successful
Install cryptography 35.0.0 successful
Installing wheel into cached location /home/jjangga/.cache/pdm/packages/pycparser-2.21-py2.py3-none-any
Install pycparser 2.21 successful
Install pyOpenSSL 21.0.0 successful
Install six 1.16.0 successful
Install selenium 4.0.0 failed
Error occurs: 
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/synchronizers.py", line 190, in install_candidate
    self.manager.install(can)
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/manager.py", line 38, in install
    installer(candidate.build(), self.environment, candidate.direct_url())
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/installers.py", line 124, in install_wheel_with_cache
    dist_info_dir = _install_wheel(
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/installers.py", line 176, in _install_wheel
    for record_elements, stream in source.get_contents():
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/installer/sources.py", line 170, in get_contents
    assert record is not None, "In {}, {} is not mentioned in RECORD".format(
AssertionError: In /tmp/pip-unpack-i08w8ctf/selenium-4.0.0-py3-none-any.whl, selenium/__init__.py is not mentioned in RECORD
Install sniffio 1.2.0 successful
Install sortedcontainers 2.4.0 successful
Install trio 0.19.0 successful
Install trio-websocket 0.9.2 successful
Install urllib3 1.26.7 successful
Install wsproto 1.0.0 successful
  Retry failed jobs
Install selenium 4.0.0 failed
Error occurs: 
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/synchronizers.py", line 190, in install_candidate
    self.manager.install(can)
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/manager.py", line 38, in install
    installer(candidate.build(), self.environment, candidate.direct_url())
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/installers.py", line 124, in install_wheel_with_cache
    dist_info_dir = _install_wheel(
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/installers.py", line 176, in _install_wheel
    for record_elements, stream in source.get_contents():
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/installer/sources.py", line 170, in get_contents
    assert record is not None, "In {}, {} is not mentioned in RECORD".format(
AssertionError: In /tmp/pip-unpack-i08w8ctf/selenium-4.0.0-py3-none-any.whl, selenium/__init__.py is not mentioned in RECORD

ERRORS:
add selenium failed:
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/opt/python@3.10/lib/python3.10/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/synchronizers.py", line 190, in install_candidate
    self.manager.install(can)
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/manager.py", line 38, in install
    installer(candidate.build(), self.environment, candidate.direct_url())
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/installers.py", line 124, in install_wheel_with_cache
    dist_info_dir = _install_wheel(
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/installers.py", line 176, in _install_wheel
    for record_elements, stream in source.get_contents():
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/installer/sources.py", line 170, in get_contents
    assert record is not None, "In {}, {} is not mentioned in RECORD".format(
AssertionError: In /tmp/pip-unpack-i08w8ctf/selenium-4.0.0-py3-none-any.whl, selenium/__init__.py is not mentioned in RECORD

Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/bin/pdm", line 8, in <module>
    sys.exit(main())
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/core.py", line 200, in main
    return Core().main(args)
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/core.py", line 153, in main
    raise cast(Exception, err).with_traceback(traceback)
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/core.py", line 148, in main
    f(options.project, options)
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/cli/commands/add.py", line 64, in handle
    actions.do_add(
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/cli/actions.py", line 222, in do_add
    do_sync(
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/cli/actions.py", line 169, in do_sync
    handler.synchronize()
  File "/home/linuxbrew/.linuxbrew/Cellar/pdm/1.10.1/libexec/lib/python3.10/site-packages/pdm/installers/synchronizers.py", line 357, in synchronize
    raise InstallationError("Some package operations are not complete yet")
pdm.exceptions.InstallationError: Some package operations are not complete yet
```

## Environment Information

```bash
# Paste the output of `pdm info && pdm info --env` below:
PDM version:        1.10.1
Python Interpreter: /home/jjangga/.pyenv/versions/3.9.7/bin/python3 (3.9)
Project Root:       /media/jjangga/SHARE/experiment/pdm
Project Packages:   /media/jjangga/SHARE/experiment/pdm/__pypackages__/3.9
{
  "implementation_name": "cpython",
  "implementation_version": "3.9.7",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "5.11.0-38-generic",
  "platform_system": "Linux",
  "platform_version": "#42~20.04.1-Ubuntu SMP Tue Sep 28 20:41:07 UTC 2021",
  "python_full_version": "3.9.7",
  "platform_python_implementation": "CPython",
  "python_version": "3.9",
  "sys_platform": "linux"
}
```

**Comment by frostming** [comment]
See [this comment](https://github.com/pdm-project/pdm/issues/719#issuecomment-961758667)

**Comment by jjangga0214** [comment]
Thanks!

**Comment by sebthom** [comment]
@frostming is there any (dirty) workaround for the time being? The selenium dependency installs fine with poetry.

**Comment by frostming** [comment]
@sebthom You can try `export PYTHONOPTIMIZ=1` to omit the AssertionError.

**Comment by frostming** [comment]
@sebthom Sorry that didn't work. The workaround is using pip:

```
pip install -t __pypackages__/3.10/lib selenium
```

**Comment by frostming** [comment]
This issue was fixed in https://github.com/SeleniumHQ/selenium/issues/10022#issuecomment-1063033886 and released in `selenium 4.1.3`
